src/fundus/scraping/scraper.py

```diff
@@ -42,27 +41,29 @@ def scrape(
 
                 return
 
-        for crawler in self.sources:
-            for article_source in crawler.fetch(batch_size):
+        for html_source in self.sources:
+            async for html in html_source.async_fetch():
                 try:
-                    extraction = self.parser(article_source.crawl_date).parse(article_source.content, error_handling)
+                    extraction = self.parser(html.crawl_date).parse(html.content, error_handling)
 
-                    if extraction_filter and extraction_filter(extraction):
-                        continue
                 except Exception as err:
                     if error_handling == "raise":
-                        error_message = f"Run into an error processing '{article_source.url}'"
+                        error_message = f"Run into an error processing article '{html.requested_url}'"
                         basic_logger.error(error_message)
                         err.args = (str(err) + "\n\n" + error_message,)
                         raise err
                     elif error_handling == "catch":
-                        yield Article(html=article_source, exception=err)
+                        yield Article(html=html, exception=err)
                         continue
                     elif error_handling == "suppress":
-                        basic_logger.info(f"Skipped {article_source.url} because of: {err!r}")
+                        basic_logger.info(f"Skipped article at '{html.requested_url}' because of: {err!r}")
                         continue
                     else:
                         raise ValueError(f"Unknown value '{error_handling}' for parameter <error_handling>'")
 
-                article = Article.from_extracted(html=article_source, extracted=extraction)
-                yield article
+                if extraction_filter and extraction_filter(extraction):
+                    basic_logger.debug(f"Skipped article at '{html.requested_url}' because of extraction filter")
+                    yield None
```

Comment:
We yield `None` here so the generator does not block when encountering an incomplete article. Since we're running in batches, blocking one generator would block the entire batch thus taking more time and fetching less HTML in parallel.

Reply:
Thanks make sense. Thanks for the explanation. Putting a short comment would properly help to remember this implementation detail.